src/main/java/inventorysetups/InventorySetupsPlugin.java

```diff
@@ -1036,10 +1029,29 @@ private boolean isNewPositionInvalid(int oldPosition, int newPosition, int size)
 
 	public List<InventorySetup> filterSetups(String textToFilter)
 	{
-		final String textToFilterLower = textToFilter.toLowerCase();
-		return inventorySetups.stream()
-			.filter(i -> i.getName().toLowerCase().contains(textToFilterLower))
-			.collect(Collectors.toList());
+    return inventorySetups.stream().filter(i -> setupContains(i, textToFilter.toLowerCase())).collect(Collectors.toList());
+	}
+
+	private static boolean setupContains(InventorySetup i, String textToFilterLower)
+	{
+		if (textToFilterLower.startsWith("item:") && textToFilterLower.length() >= 6)
+		{
+			String itemName = textToFilterLower.substring("item:".length());
+			// Find setups containing the given item name
+			return containsItem(i.getInventory(), itemName) || containsItem(i.getEquipment(), itemName)
```

Comment:
This is not checking the additional filtered items. Not sure if you want it to. It's unfortunately not a List type (which I regret a lot), so it would require a special check

Reply:
Good catch, sure, I'll add this.